compiler/plugins/input/Torch/InputConversion/ConvertTMTensorToLinalgExt.cpp

```diff
@@ -145,8 +147,36 @@ struct AttentionOpConversion
     Value collapsedResult = rewriter.create<tensor::EmptyOp>(
         loc, collapsedResultShape, elementType);
 
+    // TODO: This is a hack. This should be replaced with a simple getScale()
+    // when support for scaling is plumbed to TMTensor on the torch-mlir side.
+    // Until then, we are using the default value used in scaled dot product
+    // attention by PyTorch (most models use the default value because it makes
+    // the variance of the result of softmax 1 when the mean of Q, K is 0).
+    // We use scale = 1 / sqrt(d), where d is the head dimension.
+    // See https://paperswithcode.com/method/scaled for more details.
+    //
+    // TODO: We are currently assuming that head dimension is dim = -1. Once we
+    // have support for batch dims using more general indexing maps, we should
+    // change this and rely on more general mechanisms.
+    // TODO: We are currently not handeling dynamic shape of head dimensions at
+    // all. This is because it messes with dispatch formation. This should be
+    // fixed.
+    ArrayRef<int64_t> queryShape = op.getQueryType().getShape();
+    int64_t headDim = queryShape[queryShape.size() - 1];
+    if (headDim <= 0) {
```

Comment:
For a dynamic head dimension, should we instead check if headDim == ShapedType::kDynamic ?